heareval/tasks/nsynth_pitch.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+"""
+Pre-processing pipeline for NSynth pitch detection
+"""
+
+import os
+from pathlib import Path
+from functools import partial
+import logging
+
+import luigi
+import pandas as pd
+from slugify import slugify
+
+import heareval.tasks.config.nsynth_pitch as config
+import heareval.tasks.util.luigi as luigi_util
+
+logger = logging.getLogger("luigi-interface")
+
+# Set the task name for all WorkTasks
+luigi_util.WorkTask.task_name = config.TASKNAME
+
+
+class ExtractArchiveTrain(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.TRAIN_DOWNLOAD_URL, outfile="train-corpus.tar.gz"
+            )
+        }
+
+
+class ExtractArchiveValidation(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.VALIDATION_DOWNLOAD_URL, outfile="valid-corpus.tar.gz"
+            )
+        }
+
+
+class ExtractArchiveTest(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.TEST_DOWNLOAD_URL, outfile="test-corpus.tar.gz"
+            )
+        }
+
+
+class ConfigureProcessMetaData(luigi_util.WorkTask):
+    """
+    This config is data dependent and has to be set for each data
+    """
+
+    outfile = luigi.Parameter()
+
+    def requires(self):
+        return {
+            "train": ExtractArchiveTrain(infile="train-corpus.tar.gz"),
+            "valid": ExtractArchiveValidation(infile="valid-corpus.tar.gz"),
+            "test": ExtractArchiveTest(infile="test-corpus.tar.gz"),
+        }
+
+    @staticmethod
+    def get_rel_path(root: Path, item: pd.DataFrame) -> str:
+        # Creates the audio relative path for a dataframe item
+        audio_path = root.joinpath("audio")
+        filename = f"{item}.wav"
+        return audio_path.joinpath(filename)
+
+    @staticmethod
+    def slugify_file_name(filename: str) -> str:
+        return f"{slugify(filename)}.wav"
+
+    def get_split_metadata(self, split: str) -> pd.DataFrame:
+        logger.info(f"Preparing metadata for {split}")
+
+        # Loads and prepares the metadata for a specific split
+        split_path = Path(self.requires()[split].workdir).joinpath(f"nsynth-{split}")
+
+        metadata = pd.read_json(split_path.joinpath("examples.json"), orient="index")
+
+        # Filter out pitches that are not within the range of a standard piano
+        metadata = metadata[metadata["pitch"] >= 21]
+        metadata = metadata[metadata["pitch"] <= 108]
```

Comment:
Should we define these ranges in the config file as well? Or maybe at the top of the file with uppercase variables.

Reply:
Ya good idea, let's move this into the config.